### PR TITLE
Feat/send grid integration

### PR DESCRIPTION
**Pull request by webcoderspeed**

### CODE REVIEW

File: src/pages/api/signup.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    from: {
+      name: 'Vinci Club', email: 'email@gmail.com'
+    },
+    to: 'email@gmail.com',
+    subject: 'subject',
+    html: message,
+    templateId: process.env.CLIENT_TEMPLATE_ID,
+  };
+
+  console.log(JSON.stringify(mailOptions))
```
Comment (wa1id) [inline src/pages/api/signup.ts:29]: remove console log

File: src/pages/api/signup.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  //   referenceClub,
+  // } = req.body;
+
+  let message = 'html_template';
+
+  const mailOptions: MailDataRequired = {
+    from: {
+      name: 'Vinci Club', email: 'email@gmail.com'
+    },
+    to: 'email@gmail.com',
```
Comment (wa1id) [inline src/pages/api/signup.ts:23]: `email` should be info@vinci-club.be

`to:` should be the client email no?

File: src/pages/api/signup.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    subject: 'subject',
```
Comment (wa1id) [inline src/pages/api/signup.ts:24]: subject is: "Bedankt voor uw inschrijving bij Vinci Club"

**Review by wa1id — CHANGES_REQUESTED** [review]

File: src/pages/api/signup.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import sgMail, { MailDataRequired } from '@sendgrid/mail';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+sgMail.setApiKey(process.env.SENDGRID_API_KEY as string);
```
Comment (wa1id) [inline src/pages/api/signup.ts:4]: `SENDGRID_API`

File: src/pages/api/signup.ts (hunk 2 above)
Comment (webcoderspeed) [reply src/pages/api/signup.ts:23]: as we sending it to client then yes we have to client email at [ to: client_email ]. also we need to design the template for mail

File: src/pages/api/signup.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  const mailOptions: MailDataRequired = {
+    from: {
+      name: 'Vinci Club',
+      email: 'info@vinci-club.be',
+    },
+    to: email,
+    subject: 'Bedankt voor uw inschrijving bij Vinci Club',
+    html: message,
+    templateId: process.env.CLIENT_TEMPLATE_ID,
```
Comment (wa1id) [inline src/pages/api/signup.ts:27]: Can we set the template ID directly here? The SENDGRID_API key protects it anyways.
`d-499d3dc411d94157bb39052aa415fc53`

File: src/pages/api/signup.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  let message = 'html_template';
+
+  const mailOptions: MailDataRequired = {
+    from: {
+      name: 'Vinci Club',
+      email: 'info@vinci-club.be',
+    },
+    to: email,
+    subject: 'Bedankt voor uw inschrijving bij Vinci Club',
+    html: message,
```
Comment (wa1id) [inline src/pages/api/signup.ts:26]: What does this mean? `message` is just set to a string?

Comment (webcoderspeed) [reply src/pages/api/signup.ts:26]: sorry I forgot we are using the template. let me push the code again

File: src/pages/api/signup.ts (hunk 5 above)
Comment (webcoderspeed) [reply src/pages/api/signup.ts:27]: okay

**Pull request merged**